src/js/autocomplete.js

```diff
@@ -1,11 +1,41 @@
 'use strict'
 
+// Helper functions for handling both string and object option formats
+const getOptionText = (option) => {
+  if (option == null) return ''
+  return typeof option === 'string' ? option : (option.text || '')
+}
+
+const getOptionValue = (option) => {
+  if (option == null) return ''
+  return typeof option === 'string' ? option : (option.value || option.text || '')
+}
+
+const isObject = (value) => {
+  return value !== null && typeof value === 'object'
+}
+
+const normalizeCase = (text, config) => {
+  return config.caseSensitive ? text : text.toLowerCase()
+}
+
+const ensureStringOption = (option) => {
+  // Keep objects as-is, but convert primitives to strings to prevent breaking changes
+  return isObject(option) ? option : String(option)
+}
+
 const defaultFilterFunction = {
   start: function (token, match, config) {
-    return match.indexOf(token) === 0
+    const normalizedToken = normalizeCase(token, config)
+    return isObject(match)
+      ? normalizeCase(match.text || '', config).indexOf(normalizedToken) === 0 || normalizeCase(match.value || '', config).indexOf(normalizedToken) === 0
```

Comment:
Thanks for adding all the ` || ''` cases, that adds more robustness. I think it would be neat to move this inside the `normalizeCase` function so we don't have to repeat it this often. What do you think?